lib/browse_everything/driver/file_system.rb

```diff
@@ -36,18 +44,32 @@ def details(path, display = nil)
         return nil unless File.exist?(path)
         info = File::Stat.new(path)
         BrowseEverything::FileEntry.new(
-          Pathname.new(File.expand_path(path)).relative_path_from(Pathname.new(config[:home])),
+          make_pathname(path),
           [key, path].join(':'),
-          display || File.basename(path),
+          make_display(path, display),
           info.size,
           info.mtime,
           info.directory?
         )
       end
 
+      def make_pathname(path)
+        Pathname.new(File.expand_path(path)).relative_path_from(Pathname.new(config[:home]))
+      end
+
+      def get_file_size(path)
+        File.size(path).to_i
+      rescue
```

Comment:
It looks like `File.size()` throws `Errno::ENOENT` on non-existent files, and returns 0 for files it can't open, or special files (entries under `/dev`, for example).